mail_forward/wizards/mail_compose_message.py

```diff
@@ -1,11 +1,164 @@
 # Copyright 2024 Tecnativa - Carlos Lopez
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
-from odoo import Command, _, api, models
+from odoo import _, api, fields, models
 
 
 class MailComposeMessage(models.TransientModel):
     _inherit = "mail.compose.message"
 
+    forward_type = fields.Selection(
+        [
+            ("current_thread", "Current thread"),
+            ("another_thread", "Another thread"),
+        ],
+        default="current_thread",
+    )
+    forward_thread = fields.Reference(
+        selection="_selection_forward_thread", string="Thread to forward"
+    )
+
+    @api.model
+    def _selection_forward_thread(self):
+        # Get all models available to be selected by the user.
+        # Only consider models that support posted messages and are not transient.
+        models = (
+            self.env["ir.model"]
+            .sudo()
+            .search(
+                [("enable_forward_to_thread", "=", True)],
```

Comment:
I still think you should keep the other conditions. If the flag is forced by other means (export/import, shell), we can't allow a crash at this level.